spring-boot-project/spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/JarWriter.java

```diff
@@ -81,13 +81,14 @@ public JarWriter(File file) throws FileNotFoundException, IOException {
 	 */
 	public JarWriter(File file, LaunchScript launchScript)
 			throws FileNotFoundException, IOException {
-		FileOutputStream fileOutputStream = new FileOutputStream(file);
-		if (launchScript != null) {
-			fileOutputStream.write(launchScript.toByteArray());
-			setExecutableFilePermission(file);
+		try (FileOutputStream fileOutputStream = new FileOutputStream(file)) {
```

Comment:
This will close `fileOutputStream` before it can be written to via `this.jarOutput`. It's eventually closed when `JarWriter` is closed this is change isn't needed. I'll address this while merging.